kaldb/src/main/java/com/slack/kaldb/chunkrollover/DiskOrMessageCountBasedRolloverStrategy.java

```diff
@@ -106,27 +107,24 @@ public static long calculateDirectorySize(AtomicReference<FSDirectory> activeChu
   public static long calculateDirectorySize(FSDirectory activeChunkDirectory) {
     try {
       if (activeChunkDirectory != null && activeChunkDirectory.listAll().length > 0) {
-
-        long directorySize =
-            Arrays.stream(activeChunkDirectory.listAll())
-                .mapToLong(
-                    file -> {
-                      try {
-                        return activeChunkDirectory.fileLength(file);
-                      } catch (IOException e) {
-                        // There can be a race condition b/w the listAll which filters
-                        // pendingDeletes and then fileLength method which will throw
-                        // NoSuchFileException if the file has now become a pending delete
-                        return 0;
-                      }
-                    })
-                .sum();
-        return directorySize;
+        return SegmentInfos.readLatestCommit(activeChunkDirectory).asList().stream()
+            .mapToLong(
+                segmentCommitInfo -> {
+                  try {
+                    return segmentCommitInfo.sizeInBytes();
+                  } catch (IOException e) {
+                    return 0;
+                  }
+                })
+            .sum();
       }
+    } catch (IndexNotFoundException ignored) {
+      // no committed index found (may be brand new)
     } catch (Exception e) {
       LOG.error("Error calculating the directory size", e);
+      return -1;
     }
-    return -1;
+    return 0;
```

Comment:
We have this code comment in the class today. Is returning two different values for different exceptions helpful?


```
          // in case the method fails to calculate we return -1 so don't update the old value
          if (dirSize > 0) {
            approximateDirectoryBytes.set(dirSize);
          }
```

Reply:
I guess you need it for the test